covfee/loader.py

```diff
@@ -30,82 +32,109 @@ def cli_create_tables():
+            )
 
-        self.working_dir = self.project_spec_file.parent
-        self.projects = []
+        return covfee_app
 
-    def json_parse(self, with_spinner=True):
+    def _load_projects_json_specs_from_json_file(self, with_spinner=True) -> List[Dict]:
+        projects_json_specs: List[Dict] = []
         with Halo(
-            text=f"Parsing file {self.project_spec_file} as json..",
+            text=f"Parsing file {self._project_spec_file} as json..",
             spinner="dots",
             enabled=with_spinner,
         ) as spinner:
             try:
-                project_spec = json.load(open(self.project_spec_file))
+                project_json_specs: Dict = json.load(open(self._project_spec_file))
             except Exception as e:
                 spinner.fail(
-                    f"Error parsing file {self.project_spec_file} as JSON. Are you sure it is valid json?"
+                    f"Error parsing file {self._project_spec_file} as JSON. Are you sure it is valid json?"
                 )
                 raise e
 
-            self.projects.append(project_spec)
-            spinner.succeed(f"Read covfee file {self.project_spec_file}.")
+            projects_json_specs.append(project_json_specs)
+            spinner.succeed(f"Read covfee file {self._project_spec_file}.")
+        return projects_json_specs
 
-    def json_validate(self, with_spinner=False):
+    def _raise_exception_if_projects_json_specs_are_not_valid(
+        self, projects_json_specs: List[Dict], with_spinner=False
+    ) -> None:
         filter = AjvValidator()
-        for project_spec in self.projects:
+        for project_json_specs in projects_json_specs:
             with Halo(
-                text=f'Validating project {project_spec["name"]}',
+                text=f'Validating project {project_json_specs["name"]}',
                 spinner="dots",
                 enabled=with_spinner,
             ) as spinner:
                 try:
-                    filter.validate_project(project_spec)
+                    filter.validate_project(project_json_specs)
                 except Exception as e:
                     spinner.fail(
-                        f'Error validating project "{project_spec["name"]}".\n'
+                        f'Error validating project "{project_json_specs["name"]}".\n'
                     )
                     raise e
-                spinner.succeed(f'Project "{project_spec["name"]}" is valid.')
-
-    def json_make(self, with_spinner=False):
-        projects = []
-        for project_spec, cf in self.projects:
+                spinner.succeed(f'Project "{project_json_specs["name"]}" is valid.')
+
+    def _parse_projects_json_specs_into_covfee_app(
+        self, projects_json_specs: List[Dict], with_spinner=False
+    ) -> CovfeeApp:
+        raise NotImplementedError
+        # FIXME: ...this function was broken, but was modified to provide a skeleton of what's needed.
```

Comment:
As I wrote above, I think all of these json stuff should be removed. I would raise a `notImplementedError`